grpclb/src/test/java/io/grpc/grpclb/GrpclbLoadBalancerTest.java

```diff
@@ -2017,71 +2017,90 @@ private static Attributes eagAttrsWithToken(String token) {
   }
 
   @Test
   public void retrieveModeFromLbConfig_nullConfigUseRoundRobin() throws Exception {
-    Mode mode = retrieveModeFromLbConfig(null, channelLogger);
+    GrpclbConfig config = retrieveGrpcLbConfig(null, channelLogger);
     assertThat(logs).isEmpty();
-    assertThat(mode).isEqualTo(Mode.ROUND_ROBIN);
+    assertThat(config.getMode()).isEqualTo(Mode.ROUND_ROBIN);
+    assertThat(config.getTarget()).isNull();
   }
 
   @Test
   public void retrieveModeFromLbConfig_emptyConfigUseRoundRobin() throws Exception {
     String lbConfig = "{}";
 
-    Mode mode = retrieveModeFromLbConfig(parseJsonObject(lbConfig), channelLogger);
+    GrpclbConfig config = retrieveGrpcLbConfig(parseJsonObject(lbConfig), channelLogger);
     assertThat(logs).isEmpty();
-    assertThat(mode).isEqualTo(Mode.ROUND_ROBIN);
+    assertThat(config.getMode()).isEqualTo(Mode.ROUND_ROBIN);
+    assertThat(config.getTarget()).isNull();
   }
 
   @Test
   public void retrieveModeFromLbConfig_emptyChildPolicyUseRoundRobin() throws Exception {
     String lbConfig = "{\"childPolicy\" : []}";
 
-    Mode mode = retrieveModeFromLbConfig(parseJsonObject(lbConfig), channelLogger);
+    GrpclbConfig config = retrieveGrpcLbConfig(parseJsonObject(lbConfig), channelLogger);
     assertThat(logs).isEmpty();
-    assertThat(mode).isEqualTo(Mode.ROUND_ROBIN);
+    assertThat(config.getMode()).isEqualTo(Mode.ROUND_ROBIN);
+    assertThat(config.getTarget()).isNull();
   }
 
   @Test
   public void retrieveModeFromLbConfig_unsupportedChildPolicyUseRoundRobin()
       throws Exception {
     String lbConfig = "{\"childPolicy\" : [ {\"nonono\" : {}} ]}";
 
-    Mode mode = retrieveModeFromLbConfig(parseJsonObject(lbConfig), channelLogger);
+    GrpclbConfig config = retrieveGrpcLbConfig(parseJsonObject(lbConfig), channelLogger);
     assertThat(logs).containsExactly("DEBUG: grpclb ignoring unsupported child policy nonono");
-    assertThat(mode).isEqualTo(Mode.ROUND_ROBIN);
+    assertThat(config.getMode()).isEqualTo(Mode.ROUND_ROBIN);
+    assertThat(config.getTarget()).isNull();
   }
 
   @Test
   public void retrieveModeFromLbConfig_skipUnsupportedChildPolicy() throws Exception {
     String lbConfig = "{\"childPolicy\" : [ {\"nono\" : {}}, {\"pick_first\" : {} } ]}";
 
-    Mode mode = retrieveModeFromLbConfig(parseJsonObject(lbConfig), channelLogger);
+    GrpclbConfig config = retrieveGrpcLbConfig(parseJsonObject(lbConfig), channelLogger);
     assertThat(logs).containsExactly("DEBUG: grpclb ignoring unsupported child policy nono");
-    assertThat(mode).isEqualTo(Mode.PICK_FIRST);
+    assertThat(config.getMode()).isEqualTo(Mode.PICK_FIRST);
+    assertThat(config.getTarget()).isNull();
+  }
+
+  @Test
+  public void retrieveModeFromLbConfig_skipUnsupportedChildPolicyWithTarget() throws Exception {
+    String lbConfig = "{\"childPolicy\" : "
+        + "[ {\"nono\" : {}}, {\"pick_first\" : {\"targetName\": \"foo.google.com\"} } ]}";
```

Comment:
"targetName" should be one the same level of "childPolicy".

Reply:
done.